aten/src/THCUNN/Threshold.cu

```diff
@@ -17,7 +17,7 @@ struct ThresholdUpdateOutput
   __device__ __forceinline__ void operator()(T *out, T *in)
   {
     T x = *in;
-    *out = (x > threshold_) ? x : val_;
+    *out = (x <= threshold_) ? val_ : x;  // this order propagates NaN
```

Comment:
How does this order propagate NaN?

Reply:
Comparisons involving 'nan' always evaluate to False. :)

Reply:
Would it be nicer to have something like
```
if (isnan(x)) {
  // do something
}
...
```

Reply:
Yeah, but we don't want to sacrifice nan detection for speed. So we are only fixing the easier ones.

Reply:
How much does the NaN detection affect the speed?

Reply:
It's adding one comparison per element, so probably quite significant. :O